serverlessworkflow/modules/ROOT/pages/data-index/data-index-quarkus-extension.adoc

```diff
@@ -0,0 +1,116 @@
+= Data Index Quarkus Extension
+:compat-mode!:
+// Metadata:
+:description: Data Index Service to allow to index and query audit data in {product_name}
+:keywords: kogito, workflow, serverless, data, dataindex, data-index, index, service, extension, quarkus
+// Referenced documentation pages
+:persistence_with_postgresql_guide: xref:persistence/persistence-with-postgresql.adoc
+:data-index-core-concepts_guide: xref:data-index/data-index-core-concepts.adoc
+:getting_started_create_first_workflow_guide: xref:getting-started/create-your-first-workflow-service.adoc
+// External pages
+:kogito_sw_timeouts_showcase_embedded_example_url: {kogito_sw_examples_url}/serverless-workflow-timeouts-showcase-embedded
+:kogito_sw_timeouts_showcase_embedded_example_application_properties_url: {kogito_sw_timeouts_showcase_embedded_example_url}/src/main/resources/application.properties
+:infinispan_url: https://infinispan.org/
+:mongo_url: https://www.mongodb.com/
+:postgresql_url: https://www.postgresql.org/
+
+
+This document describes how you add the Data Index features to your runtime service. You simply need to add the Data Index extension to the runtime and
+all the audit data will be collected and stored in the database, and the graphQL endpoint will be available to execute queries and perform management operation over the workflows instances.
+
+The example described in this document is based on the link:{kogito_sw_timeouts_showcase_embedded_example_url}[`serverless-workflow-timeouts_showcase_embedded`] example application.
+
+[[data-index-ext-overview]]
+== Overview
+
+The Data Index service has been designed to store and manage audit data from different workflows that may be hosted in different runtimes.
+Communication with the service is through events that contain the audit data and the service is responsible for storing them and exposing a GRAPHQL endpoint to allow queries and maintenance operations on the different instances of Workflows.
+
+Sometimes, to avoid deploying the service separately, it could be interesting to have the audit storage functionality and the query possibility embedded in the same application.
```

Comment:
I have another suggestion as well :) 

```suggestion
In specific use cases, to avoid deploying the service separately, it could be useful to have the audit storage functionality and the query capabilities embedded in the same application.
```